Tools/CISupport/ews-build/steps.py

```diff
@@ -3531,7 +3529,7 @@ def setLayoutTestCommand(self):
             second_results_failing_tests = set(self.getProperty('second_run_failures', set()))
             list_failed_tests_with_change = sorted(first_results_failing_tests.union(second_results_failing_tests))
             if list_failed_tests_with_change:
-                self.setCommand(self.command + ['--skipped=always'] + list_failed_tests_with_change)
+                self.setCommand(self.command + ['--skip-failing-tests', '--skipped=always'] + list_failed_tests_with_change)
```

Comment:
If the bot runs the whole set of layout tests with patch, but without patch it runs with ```--skip-failing-tests``` then I think this can (in theory) cause lot of false regressions.

Example: 

1.  Run with patch: it gives timeout on a test marked as ``` [ Failure ]``` but the test has nothing to do with the patch, it is just a flaky test that is not properly marked still as ``` [ Failure Timeout ]```.
2. Re-Run with patch: it gives timeout again on this flaky test marked as ``` [ Failure ]```.
3. Re-Run without patch: that test is skipped now and is assumed to pass without patch because of ```--skip-failing-tests```
- Result: We report that the patch caused a regression when it didn't.


I think we should be consistent with this `--skip-failing-tests` flag: either we pass it on all the test steps or we don't pass it ever.

Reply:
> For the class `RunWebKitTestsRedTree()` I think it should be better to keep using `--skip-failing-tests` as that speeds up the whole testing and the repeated retries that this class does when it finds unexpected failures should be enough to filter unexpected errors caused by the test ordering and not by the patch itself.
> 
> I guess this can be done by adding a `setLayoutTestCommand()` function on `class RunWebKitTestsRedTree` that adds back the `--skip-failing-tests` parameter to the inherited `self.command`

More put this up for discussion than anything else.

`--skip-failing-tests` is, in my opinion, a super dangerous option because it makes EWS differ from post-commit testers. If we eliminated the option completely, we could rebase against the latest commit we've tested and never bother with a clean tree run at all. That's a bit of a controversial opinion, though, so this PR is going to be in limbo for longer if and until I can build consensus around that opinion.